### ISSUE TITLE
Inconsistent ordering of lock files the first time the lock file is created compared to when the lock file already exists

### ISSUE THREAD

**Issue by lesteve** [conda-incubator/conda-lock#170]
Here is a bash script that illustrates this:
```bash
env_file=/tmp/environment.yml
lock_file=/tmp/conda-lock.yml

cat << EOF > $env_file
channels:
  - conda-forge

dependencies:
  - scikit-learn
EOF

function conda_lock_multiple_times() {
    rm -rf $lock_file
    for i in $(seq 1 4); do
        conda-lock lock --mamba -p linux-64 -f $env_file --lockfile $lock_file > /dev/null 2>&1
        echo "lock file md5sum: $(md5sum $lock_file)"
    done
}

echo "first run"
conda_lock_multiple_times

echo "second run"
conda_lock_multiple_times

echo "third run"
conda_lock_multiple_times
```

Output from the script:
```
❯ bash /tmp/test.sh
first run
lock file md5sum: e0b9c5509fa6f8732f07b6ee083442ee  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
second run
lock file md5sum: 6cebd9e7f6abfcadf260df476f2bb225  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
third run
lock file md5sum: a3e0986da5b6f4afcc23399b4612b397  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
lock file md5sum: 5fe5c3b275225e02abd17fca8529c4c3  /tmp/conda-lock.yml
```

You can see that on each of the three runs, the first lock file hash, the md5sum is not consistent between each run. Once the lock file exists, the md5sum hash is consistent between runs, so there seems to be something special the first time the lock file is created. Note there is only an ordering difference, the content_hash inside the lockfile is always the same:
```
content_hash:
    linux-64: 1e4530546764a34dd77859ea637ff222053abafbdaf503fcd26a5d282b0a0391
```

Why this matters in practice: we are hoping to have human-readable diffs when we regenerate lock files (maybe too much to ask?). If we compare the diff between the first conda-lock call (when the lockfile does not exist) and the second conda-lock call (when the lockfile already exists) the ordering will be different and the diff will be hard to read. We can work-around this by generating the lockfile twice but this is a bit of a surprising behaviour.

**Comment by mariusvniekerk** [comment]
Which are the packages that move?  Both the conda and pip parts should be sorted topologically

**Comment by lesteve** [comment]
It seems like the topological sort only happens when the lockfile already exists somehow?

Here is a script with a simpler environment with only python.
```bash
env_file=/tmp/environment.yml
lock_file=/tmp/conda-lock.yml

cat << EOF > $env_file
channels:
  - conda-forge

dependencies:
  - python
EOF

rm -rf $lock_file

echo lock_file does not exist
echo ------------------------
conda-lock lock --mamba -p linux-64 -f $env_file --lockfile $lock_file > /dev/null 2>&1
grep name: $lock_file

echo
echo lock_file already exists
echo lock_file does not exist
conda-lock lock --mamba -p linux-64 -f $env_file --lockfile $lock_file > /dev/null 2>&1
grep name: $lock_file
```

Output:
```
lock_file does not exist
------------------------
  name: libuuid
  name: libzlib
  name: ca-certificates
  name: tk
  name: openssl
  name: _openmp_mutex
  name: _libgcc_mutex
  name: readline
  name: xz
  name: bzip2
  name: libgomp
  name: tzdata
  name: ld_impl_linux-64
  name: python
  name: libgcc-ng
  name: libffi
  name: libnsl
  name: sqlite
  name: zlib
  name: ncurses

lock_file already exists
------------------------
  name: _libgcc_mutex
  name: ca-certificates
  name: ld_impl_linux-64
  name: tzdata
  name: libgomp
  name: _openmp_mutex
  name: libgcc-ng
  name: bzip2
  name: libffi
  name: libnsl
  name: libuuid
  name: libzlib
  name: ncurses
  name: openssl
  name: xz
  name: readline
  name: tk
  name: zlib
  name: sqlite
  name: python
```

If I am reading this correctly the first time the packages are not sorted, but the second time they are (python at the bottom for example)?

**Comment by lesteve** [comment]
I checked with the above snippet and this has been fixed by #181, thanks a lot!

**Comment by mariusvniekerk** [comment]
I'll cut a release later today
